src/app.ts

```diff
@@ -1,26 +1,120 @@
+import { Command } from "commander";
 import { CodeTokenizer } from "./lib/codeTokenizer";
-import { Comparison } from "./lib/comparison";
-import { Summary } from "./lib/summary";
+import { Comparison, ComparisonFilterOptions } from "./lib/comparison";
+import { Matches } from "./lib/comparison.js";
+import { FilterOptions, Summary } from "./lib/summary.js";
+
+let locations: string[] = [];
+const program = new Command();
+
+// Initial program description and version
+program.version("0.0.1").description("Plagiarism detection for programming exercises");
+
+program
+  .option(
+    "-l, --language <language>",
+    "Programming language used in the submitted files.",
+    "javascript",
+  )
+  .option(
+    "-b, --base <base>",
+    "Specifies a base file. Matches with code from this file will never be reported in the output. A typical base " +
+      "file is the supplied code for an exercise.",
+  )
+  .option(
+    "-m, --maximum-fragment-count <number>",
+    "The -m option sets the maximum number of times a given fragment may appear before it is ignored. A code fragment" +
+      "that appears in many programs is probably legitimate sharing and not the result of plagiarism. With -m N " +
+      "any fragment appearing in more than N program is filtered out. This option has precedence over the -M option, " +
+      "which is set to 0.9 by default.",
+  )
+  .option(
+    "-M --maximum-fragment-percentage <float>",
+    "The -M option sets how many percent of the files the code fragment may appear before it is ignored. A fragment " +
+      "of code that appears in many programs is probably legitimate sharing and not the result of plagiarism. With " +
+      "-M N any fragment appearing in more than N percent of the files is filtered out. " +
+      "Must be a value between 0 and 1.",
+    0.9,
+  )
+  .option("-c, --comment <string>", "Comment string that is attached to the generated report")
+  .option(
+    "-n, --fragment-output-limit",
+    "Specifies how many matching fragment are shown in the result. All fragment are " +
+      "shown then this option isn't used.",
+  )
+  .option(
+    "-s, --minimum-lines-shortest <integer>",
+    "The minimum amount of lines in the shortest code fragment in a comparison before it is shown",
```

Comment:
I still don't get the -s and -S options, especially the "shortest" bit. Can you explain this to me with an example?

Reply:
When we have a pair of ranges like for example [[3, 5], [6, 11]]. The shortest of these two is the first one, we can filter the minimum amount of lines with the -s option. The minimum amount of lines in the second one can be set with the -S option.

Reply:
I'm afraid I still don't get it. I agree we need a "minimal fragment size" option: we don't display ranges shorter than that (we can discuss if both or the ranges of a couple must meet the requirement or only one).

But why would you want to filter out long fragments?

Reply:
Ok, sorry, now I get it, both are about a minimal fragment size, but you can set a different value for both entries in a couple.

I would simplify this to "minimal fragment length" and only have one option.